File: 4033-longest-subsequence-with-non-zero-bitwise-xor/longest-subsequence-with-non-zero-bitwise-xor.cpp
class Solution {
public:
    int longestSubsequence(vector<int>& nums) {
        int total = 0;
        for(int x : nums){
            total = total ^ x;
        }
        if(total != 0){
            return nums.size();
        }
        bool nonZero = false;
        for(int x :  nums){
            if(x != 0){
                nonZero = true;
            }
        }
        if(nonZero){
            return nums.size() - 1;
        }
        else{
            return 0;
        }
    }
};

File: 4033-longest-subsequence-with-non-zero-bitwise-xor/longest-subsequence-with-non-zero-bitwise-xor.py
class Solution:
    def longestSubsequence(self, nums: List[int]) -> int:
        total = 0
        for x in nums:
            total = total ^ x

        if total != 0:
            return len(nums)

        has_nonzero = False
        for x in nums:
            if x != 0:
                has_nonzero = True

        if has_nonzero:
            return len(nums) - 1

        return 0